### PR TITLE
Fix Invitation Flow

### PR DESCRIPTION
**Pull request by stephencelis**

### CODE REVIEW

File: Sources/PointFree/Account/Account.swift [hunk 1]
```diff
@@ -41,12 +41,12 @@ private func fetchAccountData<I>(
   let ownerSubscription = Current.database.fetchSubscriptionByOwnerId(user.id)
     .mapExcept(requireSome)
 
-  let owner = ownerSubscription
+  let subscription = userSubscription <|> ownerSubscription
+
+  let owner = subscription
```
Comment (stephencelis) [inline Sources/PointFree/Account/Account.swift:46]: The "owner" supplied to the endpoint should always be the subscription owner and _not_ the owner of previous, cancelled subscriptions.

File: Sources/PointFree/Account/Invite.swift [hunk 2]
```diff
@@ -88,22 +88,6 @@ let acceptInviteMiddleware: M<Tuple2<TeamInvite.Id, User?>>
         .fetchUserById(teamInvite.inviterUserId)
         .mapExcept(requireSome)
 
-      let sendInviterEmailOfAcceptance = parallel(
```
Comment (stephencelis) [inline Sources/PointFree/Account/Invite.swift:91]: Just pushed this lazy work a lil lower.

File: Sources/PointFree/Account/Invite.swift [hunk 3]
```diff
@@ -117,18 +101,36 @@ let acceptInviteMiddleware: M<Tuple2<TeamInvite.Id, User?>>
...
       }
 
-      let deleteInvite = parallel(
-        subscription
-          .flatMap { _ in Current.database.deleteTeamInvite(teamInvite.id) }
-          .run
-      )
-
-      // fire-and-forget email of acceptance and deletion of invite
-      zip2(sendInviterEmailOfAcceptance, deleteInvite).run({ _ in })
```
Comment (stephencelis) [inline Sources/PointFree/Account/Invite.swift:127]: This was sending an invite email before any kind of work was validated.

File: Sources/PointFree/Account/Invite.swift [hunk 4]
```diff
@@ -163,12 +165,41 @@ let sendInviteMiddleware =
 
       let (email, inviter) = lower(conn.data)
 
-      return Current.database.insertTeamInvite(email, inviter.id)
+      let seatsTaken = zip2(
```
Comment (stephencelis) [inline Sources/PointFree/Account/Invite.swift:168]: Sending invites was missing some validation that was only being done on the frontend. This endpoint will now check to see how many seats are available before sending another invite.

File: Sources/PointFree/Debug.swift [hunk 5]
```diff
@@ -2,22 +2,23 @@ import Either
 import Prelude
 
 extension EitherIO {
-  public func debug<B>(prefix: String = "", _ inspect: @escaping (A) -> B) -> EitherIO {
+  public func debug<B>(prefix: @autoclosure @escaping () -> String = "", _ inspect: @escaping (A) -> B) -> EitherIO {
```
Comment (stephencelis) [inline Sources/PointFree/Debug.swift:5]: Sneaking in a small, unrelated fix. This eagerness led to printing outdated info, like the database connection URL, which I noticed when troubleshooting a Heroku db.

File: Tests/PointFreeTests/AccountTests/AccountTests.swift [hunk 6]
```diff
@@ -205,6 +205,29 @@ final class AccountTests: TestCase {
     #endif
   }
 
+  func testTeam_AsTeammate_previousSubscription() {
```
Comment (stephencelis) [inline Tests/PointFreeTests/AccountTests/AccountTests.swift:208]: This test fails without the shuffling of owner logic.

File: Tests/PointFreeTests/InviteTests.swift [hunk 7]
```diff
@@ -18,7 +18,26 @@ import XCTest
 class InviteIntegrationTests: LiveDatabaseTestCase {
   override func setUp() {
     super.setUp()
-//    SnapshotTesting.record = true
+//    SnapshotTesting.isRecording = true
+  }
+
+  func testSendInvite_HappyPath() {
```
Comment (stephencelis) [inline Tests/PointFreeTests/InviteTests.swift:24]: Added a send invite happy path test.

File: Tests/PointFreeTests/InviteTests.swift [hunk 8]
```diff
@@ -18,7 +18,51 @@ import XCTest
...
+
+    Current.stripe.fetchSubscription = const(pure(.teamYearly))
+
+    let sendInvite = request(to: .invite(.send("blobber@pointfree.co")), session: .init(flash: nil, userId: inviterUser.id))
+    let conn = connection(from: sendInvite)
+
+    assertSnapshot(matching: conn |> siteMiddleware, as: .ioConn)
+  }
+
+  func testSendInvite_UnhappyPath_NoSeats() {
```
Comment (stephencelis) [inline Tests/PointFreeTests/InviteTests.swift:43]: And a test for an unhappy path.

**Review by mbrandonw — APPROVED** [review]
